analyzers/tests/SonarAnalyzer.Test/TestCases/AspNet/CallModelStateIsValid.cs

```diff
@@ -0,0 +1,248 @@
+    public string Get([Required, FromQuery] string id)                           // Compliant
+    {
+        if (!ModelState.IsValid)
+        {
+            return null;
+        }
+        return "Hello";
+    }
+}
+
+public class NonCompliantController : ControllerBase
+{
+    [HttpGet("/[controller]")]
+    public string Get([Required, FromQuery] string id)                           // Noncompliant {{ModelState.IsValid should be checked in controller actions.}}
+    //            ^^^
+    {
+        return "Hello!";
+    }
+
+    [HttpPost("/[controller]")]
+    public string Post(Movie movie)                                             // Noncompliant
+    {
+        return "Hello!";
+    }
+
+    [HttpPut("/[controller]")]
+    public string Put(Movie movie)                                              // Noncompliant
+    {
+        return "Hello!";
+    }
+
+    [HttpDelete("/[controller]")]
+    public string Delete(Movie movie)                                           // Noncompliant
+    {
+        return "Hello!";
+    }
+
+    [HttpPatch("/[controller]")]
+    public string Patch(Movie movie)                                            // Noncompliant
+    {
+        return "Hello!";
+    }
+
+    [HttpGet]
+    [HttpPost]
+    [HttpPut]
+    [HttpDelete]
+    [HttpPatch]
+    [Route("/[controller]/mix")]
+    public string Mix([Required, FromQuery, EmailAddress] string email)         // Noncompliant
+    {
+        return "Hello!";
+    }
+
+    [AcceptVerbs("GET")]
+    [Route("/[controller]/accept-verbs")]
+    public string AGet([Required] string id)                                    // Noncompliant
+    {
+        return "Hello!";
+    }
+
+    [AcceptVerbs("POST")]
+    [Route("/[controller]/accept-verbs")]
+    public string APost(Movie movie)                                            // Noncompliant
+    {
+        return "Hello!";
+    }
+
+    [AcceptVerbs("PUT")]
+    [Route("/[controller]/accept-verbs")]
+    public string APut(Movie movie)                                             // Noncompliant
+    {
+        return "Hello!";
+    }
+
+    [AcceptVerbs("DELETE")]
+    [Route("/[controller]/accept-verbs")]
+    public string ADelete(Movie movie)                                          // Noncompliant
+    {
+        return "Hello!";
+    }
+
+    [AcceptVerbs("PATCH")]
+    [Route("/[controller]/accept-verbs")]
+    public string APatch(Movie movie)                                           // Noncompliant
+    {
+        return "Hello!";
+    }
+
+    [AcceptVerbs("GET", "POST", "PUT", "DELETE", "PATCH")]
+    [Route("/[controller]/many")]
+    public string Many([Required, FromQuery, EmailAddress] string email)        // Noncompliant
+    {
+        return "Hello!";
+    }
+
+    [HttpPost("/[controller]/without-validation")]
+    public string DtoWithoutValidation(DtoWithoutValidation dto)                // Compliant, DtoWithoutValidation does not have any validation attributes.
```

Comment:
You could still post invalid json, which should set ModelState.IsValid to false.